src/main/java/uk/gov/gchq/palisade/resource/impl/DirectoryResource.java

```diff
@@ -41,20 +44,22 @@ public DirectoryResource id(final String id) {
     }
 
     public DirectoryResource parent(final ParentResource parent) {
-        Objects.requireNonNull(parent, "The parent cannot be set to null");
+        requireNonNull(parent, "The parent cannot be set to null");
         this.parent = parent;
         return this;
     }
 
     @Override
+    @Generated
     public ParentResource getParent() {
-        Objects.requireNonNull(parent, "The parent resource has not been set for this resource.");
         return parent;
     }
 
     @Override
+    @Generated
     public void setParent(final ParentResource parent) {
-        parent(parent);
+        requireNonNull(parent);
+        this.parent = parent;
     }
 
     @Override
```

Comment:
update equals, hashCode and toString methods

Reply:
OK, done